src/interface/guide/components/DamageContribution.tsx

```diff
@@ -0,0 +1,215 @@
+import styled from '@emotion/styled';
+import Spell from 'common/SPELLS/Spell';
+import { formatPercentage, formatNumber } from 'common/format';
+import GuideDataWrapper from './GuideDataWrapper';
+import DonutChart, { DonutSegment } from './DonutChart';
+import { ReactNode } from 'react';
+
+interface SpellContribution {
+  spell: Spell;
+  color: string;
+  amount: number;
+}
+
+interface Props {
+  /** Title for the chart (defaults to "Damage Contribution") */
+  title?: string;
+  /** List of spells to track with their colors */
+  spells: Array<{ spell: Spell; color: string }>;
+  /** Function to calculate damage/healing for a spell */
+  calculateContribution: (spellId: number) => number;
+  /** Color for the "Other" category */
+  otherColor?: string;
+  /** Size of the donut chart in pixels */
+  size?: number;
+  /** Inner radius ratio (0-1, where 0.5 = half the radius) */
+  innerRadiusRatio?: number;
+  /** Helper text to display below the chart */
+  helperText?: string;
+  /** Optional stat cards to display in the header */
+  stats?: ReactNode;
+}
+
+/**
+ * Displays damage/healing contribution as a donut chart with spell breakdown and legend.
+ * Automatically includes an "Other" category for untracked spells.
+ * Built using GuideDataWrapper for consistent styling.
+ *
+ * @param title - Title for the chart (defaults to "Damage Contribution")
+ * @param spells - List of spells to track with their display colors
+ * @param calculateContribution - Function that takes spellId and returns damage/healing amount
+ * @param otherColor - Color for the "Other" category (default: #666666)
+ * @param size - Size of the donut chart in pixels (default: 200)
+ * @param innerRadiusRatio - Inner radius ratio 0-1 (default: 0.6)
+ * @param helperText - Optional helper text to display below the header
+ * @param stats - Optional stat cards to display in the header
+ */
+export default function DamageContribution({
```

Comment:
i don't believe that this component is at the correct level of abstraction. that is making it very narrow & specific.

this component merges bits of analysis (injected via `calculateContribution`) with display (`Legend` layout with `DonutChart`).

what i'd suggest is:

- move the Legend stuff to `DonutChart` (can be optional with a boolean prop)
- add a helper function for building the damage segments (which can either be free-standing, or exposed like`DonutChart.DamageBreakdown`). this helper probably shouldn't require a user-supplied `calculateContribution` method, and should "just work". If the user wants to supply a custom `calculateContribution`, they should probably just build the segments themselves